app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.kt

```diff
@@ -165,8 +166,9 @@ object EventPlatformClient {
             if (!Prefs.isEventLoggingEnabled) {
                 return
             }
-            QUEUE.groupBy { it.stream }.forEach { (stream, events) ->
-                sendEventsForStream(STREAM_CONFIGS[stream]!!, events)
+            val eventsPerStream = QUEUE.groupByTo(arrayMapOf()) { it.stream }
+            for (i in 0 until eventsPerStream.size) {
```

Comment:
Why not keep using the `forEach` construct here? (and in other places where you've switched to iterating through `indices`?)